display steps and colonized planets using tokens

rzulian opened the pull request:
<img width="463" alt="screen shot 2018-09-27 at 16 56 04" src="https://user-images.githubusercontent.com/29941924/46155360-679c6680-c277-11e8-8936-b5a802ac5dcc.png">
This is displaying steps and colonised planets using tokens, that are more readable then the chars, using the shadow.
I've created a new component.
I was trying to let it being similar to the other info but with no luck, and I'm completely ignorant in css.
@coyotte508 Can you help me?

coyotte508 commented:
I don't understand what the problem is? It seems fine from the screenshot

rzulian commented:
Well, the space between the lines is different, and also the color is different.
And the tiles and boosters are now aligned in the middle :(

coyotte508 commented:
You can make the lines take more height with `line-height` (for example: `line-height: 1.5em`). 

The boosters are in the middle on chrome, it's not a new problem. It happens when the boosters are all in one line. Maybe adding an invisible element with a height of 1px and a large width would fix the problem.

I don't see any problems with the color!

coyotte508 commented:
Just noticed that you're putting element in a SVG!

I understand why, because the tokens are `<g>` tags and as suchs need to be in a  `<svg>` tags. Maybe the tokens can be changed to SVG instead (like boosters or tech tiles), or you can use individual svg tags to wrap each token.

Let me know if you have trouble, I can take over if you want.

rzulian commented:
@coyotte508 I've used svg in playerInfo and removed the StepInfo component.
I've also updated the faction info.
<img width="665" alt="screen shot 2018-09-28 at 11 56 35" src="https://user-images.githubusercontent.com/29941924/46201957-bef11400-c315-11e8-8c9e-47927ee507e3.png">

coyotte508 commented on src/components/PlayerInfo.vue:
```diff
@@ -8,15 +8,28 @@
-
-      <span style="white-space: nowrap; line-height: 1em">
+ 
+        <span style="white-space: nowrap; line-height: 1em">
```
indent

coyotte508 commented on src/components/PlayerInfo.vue:
```diff
@@ -8,15 +8,28 @@
       </span><br/>
-      <span style="line-height: 1em" v-if="hasPlanets">Colonized: <span v-for="(count, planet, index) in player.ownedPlanetsCount" :key="planet" :class="{'ml-2': index > 0}"><i :class="['planet', planet]" :key="planet" /> {{count}} </span></span>
+      <span style="line-height: 1em" v-if="hasPlanets">Colonized: 
+        <span v-for="(count, planet, index) in player.ownedPlanetsCount"  :key="planet"  :class="'ml-2'">
```
No need to use `:class` as there's no 'code', you can just do `class="ml-2"`. To make it consistent with the steps, use `mr-2` instead so there's no margin after "Colonized: ".

rzulian replied:
ok

coyotte508 commented on src/components/PlayerInfo.vue:
```diff
@@ -8,15 +8,28 @@
+              <circle :cx="8" :cy="8" :r="6"  :class="['player-token', 'planet-fill', planet]" />
+            </svg>
+          </span> 
+          <span>{{i}}</span>
```
Maybe add a space just before `{{i}}`, to add a bit of space between the token and the number?

Also, the same comment applies for colonized planets.

rzulian replied:
A space before is adding to much space. Can Work for steps but not for colonized planets, because you are unsure to which one the number is referring to

rzulian replied:
I've added a pixel to the svg container
<img width="312" alt="screen shot 2018-09-28 at 14 00 24" src="https://user-images.githubusercontent.com/29941924/46207235-f0261000-c326-11e8-944a-1937368c8a34.png">

coyotte508 commented on src/components/PlayerInfo.vue:
```diff
@@ -111,6 +124,18 @@ export default interface PlayerInfo {
 </script>
 
 <style lang="scss">
+ .player-token {
```
indent

coyotte508 commented on src/components/PlayerInfo.vue:
```diff
@@ -111,6 +124,18 @@ export default interface PlayerInfo {
+}
+
+ .content {
+      font-family: arial;
```
indent

coyotte508 commented on src/data/factions.ts:
```diff
@@ -93,7 +93,11 @@ export function factionDesc(faction: Faction) {
-      ${[0,1,2,3].map(i => `<span class="ml-2">${planetsWithSteps(p, i).map(p => `<i class="planet ${p}"></i>`).join(" ")} ${i}</span>`).join(" ")}
+      ${[0,1,2,3].map(i => `<span class="ml-2">${planetsWithSteps(p, i).map(p => 
+        `<svg width="15" height="20" viewbox="0 0 15 15" >
+          <circle cx="7" cy="6" r="6"  class="player-token planet-fill ${p}" />
```
Is it normal that cx/cy are different than in PlayerInfo (7,6 instead of 8,8)?

rzulian replied:
Definitely css is not my best skill ;)
I had to do this because if not I was having 
<img width="338" alt="screen shot 2018-09-28 at 13 57 08" src="https://user-images.githubusercontent.com/29941924/46207149-9faeb280-c326-11e8-807d-263004e8bc1d.png">

coyotte508 commented on src/components/PlayerInfo.vue:
```diff
@@ -9,23 +9,23 @@
       <span style="line-height: 1em" v-if="hasPlanets">Colonized: 
-        <span v-for="(count, planet, index) in player.ownedPlanetsCount"  :key="planet"  :class="'ml-2'">
-          <svg width="15" height="20" viewbox="0 0 15 15" >
+        <span v-for="(count, planet, index) in player.ownedPlanetsCount"  :key="planet"  class="'mr-2'">
```
You can remove the quotes, `class="mr-2"` instead of `class="'mr-2'"`.

The `:` in vue js is treating the content as a variable, so that means using quotes around stings. Without the `:`, the content is just a string, there's no need to add quotes around it.
